### PR TITLE
[DOCS] Reformat script query

### PR DESCRIPTION
**Pull request by jrodewig**
Rewrites the `script` query to use the new query format.

This creates separate sections for the example request and parameters

This is part of #40977, an effort to standardize documentation for query types.

### Preview
http://elasticsearch_44882.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/query-dsl-script-query.html

### CODE REVIEW

File: docs/reference/query-dsl/script-query.asciidoc [hunk 1]
```diff
@@ -4,12 +4,16 @@
 <titleabbrev>Script</titleabbrev>
 ++++
 
-A query allowing to define
-<<modules-scripting,scripts>> as queries. They are typically used in a filter
-context, for example:
+
+Accepts another query as a <<modules-scripting-using,script>>. `script` queries
+are typically used in a <<query-filter-context,filter context>>.
+
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/script-query.asciidoc]: > Accepts another query as a <<modules-scripting-using,script>>

It doesn't really accept another query.  I would rather say as it was before or something like: 

"`script_query` is used to filter documents based on the provided script. They are typically used in a filter context."

File: docs/reference/query-dsl/script-query.asciidoc [hunk 2]
```diff
@@ -25,18 +29,29 @@ GET /_search
...
 
-Scripts are compiled and cached for faster execution. If the same script
-can be used, just with different parameters provider, it is preferable
-to use the ability to pass parameters to the script itself, for example:
+[[script-top-level-params]]
+==== Top-level parameters for `script`
+
+`script`::
+(Required, <<modules-scripting-using, script object>>) Contains a script to run
+as a query.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/script-query.asciidoc]: May be worth to add that the script must return a boolean value, `true` or `false`.

**Review by mayya-sharipova — APPROVED** [review]
Thanks @jrodewig

File: docs/reference/query-dsl/script-query.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/script-query.asciidoc]: Thanks @mayya-sharipova! Fixed with 08bc253.

File: docs/reference/query-dsl/script-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/script-query.asciidoc]: Added with 021a9cc.

**Pull request merged**